metadata-ingestion/src/datahub/ingestion/source/s3/config.py

```diff
@@ -66,6 +74,11 @@ class DataLakeSourceConfig(DatasetSourceConfigMixin, PathSpecsConfigMixin):
         description="Either a boolean, in which case it controls whether we verify the server's TLS certificate, or a string, in which case it must be a path to a CA bundle to use.",
     )
 
+    number_of_files_to_sample: int = Field(
+        default=100,
+        description="Number of files to list to sample for schema inference. This will be ignored if sample_files is set to False in the pathspec.",
+    )
```

Comment:
Since we literally use the s3 connector for gcs, should we try to tie the two configs closer together, perhaps with a mixin? Could include `max_rows`, `path_specs` (well, thru inheritance), `aws_config` for example